src/super_gradients/training/models/detection_models/sliding_window_detection_forward_wrapper.py

```diff
@@ -0,0 +1,395 @@
+from typing import Optional, List
+from functools import lru_cache
+
+import torch
+from torch import nn
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.processing_factory import ProcessingFactory
+from super_gradients.module_interfaces import HasPredict
+from super_gradients.training.models import CustomizableDetector
+from super_gradients.training.utils.predict import ImagesDetectionPrediction
+from super_gradients.training.pipelines.pipelines import SlidingWindowDetectionPipeline
+from super_gradients.training.processing.processing import Processing, ComposeProcessing, DetectionAutoPadding
+from super_gradients.training.utils.media.image import ImageSource
+import torchvision
+from super_gradients.training.utils.detection_utils import DetectionPostPredictionCallback
+
+
+class SlidingWindowInferenceDetectionWrapper(HasPredict, nn.Module):
+    """
+    Implements a sliding window inference wrapper for a customizable detector.
+
+    Parameters:
```

Comment:
A docstring is not following `:param param_name: ` pattern that we use elsewhere